tests/std/tests/GH_004275_seeking_fancy_iterators/test.cpp

```diff
@@ -0,0 +1,454 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// some portions of this file are derived from libc++'s test files:
+// * support/min_allocator.h
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <iterator>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+using namespace std;
+
+template <class T, class = enable_if<false>>
+class min_pointer;
+template <class T, class ID>
+class min_pointer<const T, ID>;
+template <class ID>
+class min_pointer<void, ID>;
+template <class ID>
+class min_pointer<const void, ID>;
+template <class T>
+class min_allocator;
+
+template <class ID>
+class min_pointer<const void, ID> {
+    const void* ptr_;
+
+public:
+    min_pointer() = default;
+    min_pointer(nullptr_t) noexcept : ptr_(nullptr) {}
+    template <class T>
+    min_pointer(min_pointer<T, ID> p) noexcept : ptr_(p.ptr_) {}
+
+    explicit operator bool() const {
+        return ptr_ != nullptr;
+    }
+
+    friend bool operator==(min_pointer x, min_pointer y) {
+        return x.ptr_ == y.ptr_;
+    }
+    friend bool operator!=(min_pointer x, min_pointer y) {
+        return !(x == y);
+    }
+    template <class U, class XID>
+    friend class min_pointer;
+};

```

Comment:
No change requested: I observe that `min_pointer<const void, ID>` lacks `constexpr` when the other specializations have it. This appears to be the case in libc++'s original code, so maybe they were trying to do something special there.